aesd-char-driver/aesd-circular-buffer.c

```diff
@@ -97,3 +143,21 @@ void aesd_circular_buffer_init(struct aesd_circular_buffer *buffer)
 {
     memset(buffer,0,sizeof(struct aesd_circular_buffer));
 }
+
+void aesd_circular_buffer_clean(struct aesd_circular_buffer *buffer)
+{
+    uint8_t index;
+    struct aesd_buffer_entry *entry;
+
+    AESD_CIRCULAR_BUFFER_FOREACH(entry, buffer, index)
+    {
+        if(entry->buffptr != NULL)
+        {
+#ifdef __KERNEL__
+            kfree(entry->buffptr);
```

Comment:
@Mich2899 I'd move the free logic into the applicatoin rather than this file.  Because the application provides the buffer it will know whether it was dynamically allocated.